elytron/src/main/java/org/wildfly/extension/elytron/JdbcRealmDefinition.java

```diff
@@ -635,6 +644,32 @@ public void uninject() {
                     .install();
         }
 
+        private static KeyMapper resolveKeyMappers(OperationContext context, ModelNode authenticationQueryNode) throws OperationFailedException {
+            KeyMapper keyMapper = null;
+
+            for (String name : authenticationQueryNode.keys()) {
+                ModelNode propertyNode = authenticationQueryNode.require(name);
+
+                if (!propertyNode.isDefined()) {
+                    continue;
+                }
+
+                PasswordMapperObjectDefinition mapperResource = PrincipalQueryAttributes.SUPPORTED_PASSWORD_MAPPERS.get(name);
+
+                if (mapperResource == null) {
+                    continue;
+                }
+
+                try {
+                    keyMapper = mapperResource.toPasswordKeyMapper(context, propertyNode);
+                } catch (InvalidKeyException e) {
+                    throw new OperationFailedException("Invalid key type.", e);
+                }
+            }
+
+            return keyMapper;
+        }
+
```

Comment:
I added the Mapper validation, but I don't know if we still need to have this code on we can just remove the `InvalidKeyException` from the interface, and call this without those validations (both `if`). I guess that we can keep the `throw new OperationFailedException("Invalid key type.", e);` only at the `checkKeyMappers`. Or do we need those validations on Runtime time as well?

Reply:
Can you move this method back where it was? AFAICT it doesn't need to move, and moving it makes this diff very hard to read.

Reply:
Unless @fjuma disagrees, +1 to getting rid of 'throws InvalidKeyException'. None of the impls throw it.

If we keep it we should fix the i18 of this OFE. Exceptions should not have hard coded English messages.

Reply:
+1 for getting rid of the `throws InvalidKeyException`.